include/gridtools/boundary_conditions/apply_gpu.hpp

```diff
@@ -185,96 +181,78 @@ namespace gridtools {
                         }
                     }
                 }
+            }
 
+            dim3 block_size() const {
+                dim3 b(0, 0, 0);
                 for (int i = 0; i < 3; ++i) {
                     for (int j = 0; j < 3; ++j) {
                         for (int k = 0; k < 3; ++k) {
                             if (i != 1 or j != 1 or k != 1) {
-                                configuration[0] = max(configuration[0], sizes[i][j][k].max());
-                                configuration[1] = max(configuration[1], sizes[i][j][k].median());
-                                configuration[2] = max(configuration[2], sizes[i][j][k].min());
+                                b.x = std::max(b.x, (decltype(b.x))sizes[i][j][k].max());
+                                b.y = std::max(b.y, (decltype(b.y))sizes[i][j][k].median());
+                                b.z = std::max(b.z, (decltype(b.z))sizes[i][j][k].min());
                             }
                         }
                     }
                 }
+                assert((b.x > 0 || b.y > 0 || b.z > 0) && "all boundary extents are empty");
```

Comment:
I don't fully understand what's going on here. Just wondering if in a generic context a boundary condition might be called with 0 extents and we want to silently do nothing?

Reply:
I have taken the assertion from @mbianco’s implementation (see line 332 of the file), no idea if this assertion can be ever triggered by a user!

Reply:
Ok, I thought it is new.